crankshaft-engine/src/task/resources.rs

```diff
@@ -303,4 +341,56 @@ mod test {
         assert_eq!(tes.backend_parameters, None);
         assert_eq!(tes.backend_parameters_strict, None);
     }
+
+    #[test]
+    fn gpu_creates_device_request() {
```

Comment:
It's a bit strange to have Docker-specific testing in this module, but I see there's TES and Docker impls above that make separating it into a Docker-specific module beyond the scope of this PR.